finagle-mysql/src/main/scala/com/twitter/finagle/mysql/Request.scala

```diff
@@ -258,6 +231,20 @@ case class ExecuteRequest(
     }
 }
 
+object ExecuteRequest {
+  def apply(stmtId: Int, params: IndexedSeq[Parameter] = IndexedSeq.empty, hasNewParams: Boolean = true, flags: Byte = 0) = {
```

Comment:
Okay re: line length. 

Putting it in the constructor will be messy because of how Scala does constructors. Every parameter to class is accessible in its entire body, and also it's immutable. We will have something like:

```scala
class ExecuteRequest(..., params: Seq[Parameter], ...) {
  val sanitizedParams = sanitize(params)

  // but params is still accessible here, what if we accidentally use that?
}
```